ui/src/app/settings/containers/Users/Users.test.js

```diff
@@ -1,22 +1,45 @@
-import { shallow } from "enzyme";
+import { mount } from "enzyme";
 import React from "react";
-import { Users } from "./Users";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+import { MemoryRouter } from "react-router-dom";
+import Users from "./Users";
+
+const mockStore = configureStore();
 
 describe("Users", () => {
   it("can render", () => {
-    const wrapper = shallow(
-      <Users
-        authUser={{
-          email: "test@example.com",
-          first_name: "",
-          global_permissions: ["machine_create"],
-          id: 1,
-          is_superuser: true,
-          last_name: "",
-          sshkeys_count: 0,
-          username: "admin"
-        }}
-      />
+    const user = {
+      email: "test@example.com",
+      first_name: "",
+      global_permissions: ["machine_create"],
+      id: 1,
+      is_superuser: true,
+      last_name: "",
+      sshkeys_count: 0,
+      username: "admin"
+    };
+    const store = mockStore({
+      auth: {
+        user
+      },
+      users: {
+        items: [
+          {
+            user
+          }
+        ]
+      }
+    });
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/settings/users", key: "testKey" }]}
+        >
+          <Users />
+        </MemoryRouter>
+      </Provider>
     );
     expect(wrapper).toMatchSnapshot();
```

Comment:
Should we just snapshot the component we are testing so that the provider, router etc. aren't in the snapshot? It would make the snapshot a lot more useful maybe. E.g. `expect(wrapper.find('Users')).toMatchSnapshot();`.